### ISSUE TITLE
demo.py arg error about NQ

### ISSUE THREAD

**Issue by yuchenlin** [AkariAsai/learning_to_retrieve_reasoning_paths#11]
Hi Akari,
Thanks for the great repo. I had an error when trying to use the demo for the natural questions' model.  I followed the tip to rename `demo.py` in the previous example for running eval of NQ.

I am running 
```bash
python demo.py \
--graph_retriever_path models/nq/selector/pytorch_model.bin \
--reader_path models/nq/reader/ \
--tfidf_path models/nq_models/tfidf_retriever/wiki_20181220_nq_hyper_linked-tfidf-ngram=2-hash=16777216-tokenizer=simple.npz \
--db_path models/nq_models/wiki_db/wiki_20181220_nq_hyper_linked.db \
--bert_model bert-base-uncased --do_lower_case --tfidf_limit 20 --eval_batch_size 4 --pruning_by_links \
--beam_graph_retriever 8 --max_para_num 2000 --use_full_article 
```

And got the errors:
- `demo.py: error: ambiguous option: --bert_model could match --bert_model_graph_retriever, --bert_model_sequential_sentence_selector`. 

- `demo.py: error: unrecognized arguments: --use_full_article` （i assumed it was the `bert_model_sequential_sentence_selector`)

- `FileNotFoundError: [Errno 2] No such file or directory: 'models/nq/selector/pytorch_model.bin'` (I removed the `--use_full_article`)  I think this should be `models/nq_models/graph_retrieverpytorch_model.bin`?

- `--reader_path models/nq/reader/` seems not correct? I think this should be `models/nq_models/reader`

**Comment by yuchenlin** [comment]
also for the `--quiet`

`demo.py: error: unrecognized arguments: --quiet`

**Comment by yuchenlin** [comment]
I managed to run the demo by ignoring the two args
```
python demo.py \
--graph_retriever_path models/nq_models/graph_retriever/pytorch_model.bin \
--reader_path models/nq_models/reader \
--sequential_sentence_selector_path models/nq_models/reader/pytorch_model.bin \
--tfidf_path models/nq_models/tfidf_retriever/wiki_20181220_nq_hyper_linked-tfidf-ngram=2-hash=16777216-tokenizer=simple.npz \
--db_path models/nq_models/wiki_db/wiki_20181220_nq_hyper_linked.db \
--do_lower_case --max_para_num 200 \
--tfidf_limit 20 --pruning_by_links
```

**Comment by yuchenlin** [comment]
Ah, I just figured out. I missed the tip that I should simply replace the filename from `eval_main.py` to `demo.py` in the example of running the evaluation on NQ.  Never mind! 

Thanks.

**Comment by yuchenlin** [comment]
It seems to still have an error about `demo.py: error: ambiguous option: --bert_model could match --bert_model_graph_retriever, --bert_model_sequential_sentence_selector`

**Comment by yuchenlin** [comment]
I finally run the demo with 

```
python demo.py \
--graph_retriever_path models/nq_models/graph_retriever/pytorch_model.bin \
--reader_path models/nq_models/reader \
--tfidf_path models/nq_models/tfidf_retriever/wiki_20181220_nq_hyper_linked-tfidf-ngram=2-hash=16777216-tokenizer=simple.npz \
--db_path models/nq_models/wiki_db/wiki_20181220_nq_hyper_linked.db \
--bert_model_sequential_sentence_selector bert-base-uncased --do_lower_case --tfidf_limit 20 --eval_batch_size 4 --pruning_by_links \
--beam_graph_retriever 8 --max_para_num 2000
```

Not sure if it is correct. Thanks!

**Comment by hassyGo** [comment]
Hi @yuchenlin 

Thank you for your interest in our paper and the code.
For the `--bert_model` option, yes, your suggestion is correct, but let me add one note.

We need to specify the BERT types for the graph retriever model (for all the datasaets) and the sentence selector model (only for HotpotQA).
The former can be specified here:
https://github.com/AkariAsai/learning_to_retrieve_reasoning_paths/blob/master/demo.py#L105
and the latter can be specified here:
https://github.com/AkariAsai/learning_to_retrieve_reasoning_paths/blob/master/demo.py#L110
, so in that sense your command specifies the `--bert_model_sequential_sentence_selector` option, and it is not used for NQ.
In other words, there is no effect by specifying the `--bert_model_sequential_sentence_selector` option.
The default values are set in the code, and now your command works.

The reason why you do not face the first error is that, there is no `--bert_model` option as shown in the error message `error: ambiguous option: --bert_model could match --bert_model_graph_retriever, --bert_model_sequential_sentence_selector`, and removing the `--bert_model` option works.

We will update README to replace all the `--bert_model` options in the example commands.

Thank you for pointing it out!

**Comment by yuchenlin** [comment]
Hi Akari,

Thanks for your prompt help. Just wanna re-confirm that for running demo.py on top of NQ models, I should run: 

```
python demo.py \
--graph_retriever_path models/nq_models/graph_retriever/pytorch_model.bin \
--reader_path models/nq_models/reader \
--tfidf_path models/nq_models/tfidf_retriever/wiki_20181220_nq_hyper_linked-tfidf-ngram=2-hash=16777216-tokenizer=simple.npz \
--db_path models/nq_models/wiki_db/wiki_20181220_nq_hyper_linked.db \
--do_lower_case --tfidf_limit 20 --eval_batch_size 4 --pruning_by_links \
--beam_graph_retriever 4 --max_para_num 100
```

Note that I removed `--bert_model` and `--use_full_article`. And I changed the path of `--graph_retriever_path` and `--reader_path` as above.

**Comment by AkariAsai** [comment]
Sorry for our late response! For the demo, it should work. Please let us know if you find any issues.
